src/app/clusters/time-synchronization-server/time-synchronization-cluster.cpp

```diff
@@ -238,49 +204,258 @@ static bool emitMissingTrustedTimeSourceEvent(EndpointId ep)
     return true;
 }
 
-TimeSynchronizationServer TimeSynchronizationServer::sTimeSyncInstance;
+} // namespace TimeSynchronization
 
-TimeSynchronizationServer & TimeSynchronizationServer::Instance()
+#if TIME_SYNC_ENABLE_TSC_FEATURE
+void OnDeviceConnectedWrapper(void * context, Messaging::ExchangeManager & exchangeMgr, const SessionHandle & sessionHandle)
 {
-    return sTimeSyncInstance;
+    TimeSynchronizationCluster * timeSynchronization = reinterpret_cast<TimeSynchronizationCluster *>(context);
+    timeSynchronization->OnDeviceConnectedFn(exchangeMgr, sessionHandle);
 }
 
-TimeSynchronizationServer::TimeSynchronizationServer() :
+void OnDeviceConnectionFailureWrapper(void * context, const ScopedNodeId & peerId, CHIP_ERROR error)
+{
+    TimeSynchronizationCluster * timeSynchronization = reinterpret_cast<TimeSynchronizationCluster *>(context);
+    timeSynchronization->OnDeviceConnectionFailureFn();
+}
+#endif
+
+void OnPlatformEventWrapper(const DeviceLayer::ChipDeviceEvent * event, intptr_t ptr)
+{
+    TimeSynchronizationCluster * timeSynchronization = reinterpret_cast<TimeSynchronizationCluster *>(ptr);
+    timeSynchronization->OnPlatformEventFn(*event);
+}
+
+void OnTimeSyncCompletionWrapper(void * context, TimeSourceEnum timeSource, GranularityEnum granularity)
+{
+    TimeSynchronizationCluster * timeSynchronization = reinterpret_cast<TimeSynchronizationCluster *>(context);
+    timeSynchronization->OnTimeSyncCompletionFn(timeSource, granularity);
+}
+
+void OnFallbackNTPCompletionWrapper(void * context, bool timeSyncSuccessful)
+{
+    TimeSynchronizationCluster * timeSynchronization = reinterpret_cast<TimeSynchronizationCluster *>(context);
+    timeSynchronization->OnFallbackNTPCompletionFn(timeSyncSuccessful);
+}
+
+TimeSynchronizationCluster::TimeSynchronizationCluster(
+    EndpointId endpoint, const BitFlags<TimeSynchronization::Feature> features,
+    TimeSynchronization::Attributes::SupportsDNSResolve::TypeInfo::Type supportsDNSResolve, TimeZoneDatabaseEnum timeZoneDatabase,
+    TimeSynchronization::TimeSourceEnum timeSource) :
+    DefaultServerCluster({ endpoint, TimeSynchronization::Id }),
+    mFeatures(features), mSupportsDNSResolve(supportsDNSResolve), mTimeZoneDatabase(timeZoneDatabase), mTimeSource(timeSource),
 #if TIME_SYNC_ENABLE_TSC_FEATURE
     mOnDeviceConnectedCallback(OnDeviceConnectedWrapper, this),
     mOnDeviceConnectionFailureCallback(OnDeviceConnectionFailureWrapper, this),
 #endif
     mOnTimeSyncCompletion(OnTimeSyncCompletionWrapper, this), mOnFallbackNTPCompletion(OnFallbackNTPCompletionWrapper, this)
 {}
 
-void TimeSynchronizationServer::AttemptToGetFallbackNTPTimeFromDelegate()
+CHIP_ERROR TimeSynchronizationCluster::ReadTrustedTimeSource(AttributeValueEncoder & encoder)
```

Comment:
A separate method just makes the code larger. Squash these trivial methods into the switch.